### PR TITLE
api: proper GIF operations

### LINKED ISSUE #41: Operations on GIF images do not preserve its frames.
If we operate on a GIF, all but the first frame will be lost, thus being not animated anymore.

### PR DESCRIPTION
**Pull request by omelkonian**
* Adds custom behaviour for operations on GIF images, as `PIL.Image`
  operations do not work out-of-the-box (closes #41).

* Additionally passes `save_all=True` as a keyword arguement on
  `PIL.Image.save` when serving the file over HTTP.

* Bumps `Pillow`'s version to 3.4, where they introduce functionality
  for creating GIF images out of individual frame images.

* Updates tests to cover GIF operations.

Signed-off-by: Orestis Melkonian <melkon.or@gmail.com>

### CODE REVIEW

**Comment by egabancho** [comment]
can you rebase?

File: flask_iiif/api.py [hunk 1]
```diff
@@ -379,7 +384,12 @@ def serve(self, image_format="png", quality=90):
         image_buffer = BytesIO()
         # transform `image_format` is lower case and not equals to jpg
         cleaned_image_format = self._prepare_for_output(image_format)
-        self.image.save(image_buffer, cleaned_image_format, quality=quality)
+        save_kwargs = dict(quality=quality)
+
+        if self.image.format == 'GIF':
```
Comment (drjova) [inline flask_iiif/api.py]: just this one, are we sure is always capital `GIF`?

Comment (omelkonian) [inline flask_iiif/api.py]: I will capitalize, to be 100% sure.

**Comment by omelkonian** [comment]
@egabancho @drjova Please do not merge yet, as I will update it with a solution for the rest of the operations.

File: flask_iiif/api.py (hunk 1 above)
Comment (greut) [inline flask_iiif/api.py]: https://github.com/python-pillow/Pillow/blob/0834e9491e321ba3d7bd0f8a2409213bc4ecd214/PIL/GifImagePlugin.py#L48

**Comment by omelkonian** [comment]
@drjova ping

**Pull request closed without merging**